backend/apps/owasp/models/snapshot.py

```diff
@@ -0,0 +1,38 @@
+"""OWASP app snapshot models."""
+
+from django.db import models
+
+
+class Snapshot(models.Model):
+    """Model representing a snapshot of data processing."""
+
+    class Status(models.TextChoices):
+        PENDING = "pending", "Pending"
+        PROCESSING = "processing", "Processing"
+        COMPLETED = "completed", "Completed"
+        ERROR = "error", "Error"
+
+    start_at = models.DateTimeField()
+    end_at = models.DateTimeField()
+    status = models.CharField(max_length=20, choices=Status.choices, default=Status.PENDING)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    error_message = models.TextField(blank=True)
+
+    # Many-to-Many relationships
+    new_chapters = models.ManyToManyField("owasp.Chapter", related_name="snapshots", blank=True)
+    new_projects = models.ManyToManyField("owasp.Project", related_name="snapshots", blank=True)
+    new_issues = models.ManyToManyField("github.Issue", related_name="snapshots", blank=True)
+    new_releases = models.ManyToManyField("github.Release", related_name="snapshots", blank=True)
+    new_users = models.ManyToManyField("github.User", related_name="snapshots", blank=True)
+
+    class Meta:
+        ordering = ["-start_at"]
+        indexes = [
+            models.Index(fields=["status"]),
+            models.Index(fields=["start_at", "end_at"]),
+        ]
```

Comment:
Remove these and add `db_table` and `verbose_name_plural` that we use for in other models.